### ISSUE TITLE
Does not handle the teleport trap gracefully

### ISSUE THREAD

**Issue by Debuggernaut** [Debuggernaut/LucidNightmareHelper#4]
It's possible to get your map into a totally screwed-up state through the de-duplication code if you hit the teleport trap repeatedly

**Comment by Allenone1** [comment]
Any update on this? I've restarted the Endless halls 2 times and keep hitting this issue.

**Comment by Debuggernaut** [comment]
Ah, kinda, I was tinkering with some ideas on the eng/PR-4 branch, you can check it out.  It adds a button you can hit if you know you just got teleported (because you come out of a rock wall, or pop into a room with a different wall layout)

I think your easiest bet right now might be to use this addon in concert with this guy's website:

https://nightswimmer.github.io/EndlessHalls/

**Comment by Debuggernaut** [comment]
Hey @Allenone1 , I just pushed my changes out to a release here: https://github.com/Debuggernaut/LucidNightmareNavigator/releases/tag/ManualTeleport

It still sucks, but it's not quite as bad.  At least you won't have to scrap your map and start over, haha

**Comment by Allenone1** [comment]
Great thanks!
ill try it out, already ran through and had my map go to shit 4 times hoping to get good rng where it doesn't brick. I don't care if it sucks as long as it works, haha.

**Comment by Debuggernaut** [comment]
Yeah.. on average even for me it takes about 2 hours to run through the whole maze with this addon

I was able to run through and build a fully-functional map in LucidNightmareNavigator every time I did a test run, probably 5 times in total, but it definitely took forever.  Thankfully I managed to find a map with a particularly nasty teleport trap today, and used that to test out the new "I just got ported!" button.  I think on previous runs I'd just been lucking out and not getting a completely broken layout due to the teleport, or maybe I was just zoned out and rebuilding the map one square at a time repeatedly..

**Comment by Allenone1** [comment]
Mine have been pretty quick. just finished my run after downloading the new release you mentioned. ~40 minutes. I think i just got good rng with the teleport traps on this one.

side note: would be neat if it said in chat once you've marked everything. I think I continued like 10-15 more rooms after marking the last orb/rune without noticing.

Thanks for your work! :)
